rio_cogeo/cogeo.py

```diff
@@ -25,6 +25,18 @@
 IN_MEMORY_THRESHOLD = int(os.environ.get("IN_MEMORY_THRESHOLD", 10980 * 10980))
 
 
+@contextmanager
+def TemporaryRasterFile(fname, suffix=".tmp"):
```

Comment:
If you don't need `tempfile` then there is no need to pre-create and then close the file. So this might as well read

```python
tmp_name = fname + suffix
try:
    yield tmp_name
finally:
    os.remove(tmp_name)
```

This would clobber existing temp files, which is probably fine. If you want to introduce some entropy and make sure you are writing to a new file every time, I would use

```python
f_dir = os.path.dirname(os.path.realpath(fname))
f_prefix, f_suffix = os.path.splitext(fname)
tmp_file = tempfile.NamedTemporaryFile(
    dir=f_dir, 
    suffix=f_suffix + '.tmp',
    prefix=f_prefix + '_', 
    delete=False
)
tmp_file.close()  # to make sure we can open it again on Windows systems
try:
    yield tmp_file.name
finally:
   os.remove(tmp_file.name)
```

This should create temporary files with the pattern `raster_filename_<random hex>.tif.tmp` (but I didn't test it).

Reply:
I like the idea of still using `tempfile.NamedTemporaryFile` (to test if we can write file) 👌 

> raster_filename_<random hex>.tif.tmp

having `random hex` + `.tmp` is maybe too much, Maybe we can just rely on having `random hex` + `.tif` in the output directory